include/emp/base/error.hpp

```diff
@@ -14,9 +14,44 @@
 
 #include <iostream>
 #include <string>
+#include <sstream>
 
 namespace emp {
 
+/// TDEBUG should trigger its EMP equivalent.
+#ifdef TDEBUG
+#define EMP_TDEBUG 1
+#endif
+
+// If we're testing, do NOT abort, and instead save all data so it can be tested
+#ifdef EMP_TDEBUG
+  struct ErrorInfo {
+    std::string filename;
+    size_t line_num;
+    std::string output;
+  };
+
+  ErrorInfo error_info;
+  bool error_thrown = false;
+
+  void error_clear() { emp::error_thrown = false; }
+
+  template <typename... Ts>
+  void trigger_emp_error(std::string filename, size_t line, Ts &&... args) {
+    std::cout << "Would-be fatal error (In " << filename << " line " << line
+              <<  "): ";
+    (std::cout << ... << args);
+    std::cout << std::endl;
+    std::stringstream tmp_stream;
+    (tmp_stream << ... << args);
+    emp::error_info.filename = filename;
+    emp::error_info.line_num = line;
+    emp::error_info.output = tmp_stream.str();
+    emp::error_thrown = true;
```

Comment:
I might change verbage from “thrown” to “triggered” or “occurred” because thrown makes me think of throwing and catching exceptions.